planning/planning_test_utils/src/planning_interface_test_manager.cpp

```diff
@@ -401,14 +401,24 @@ void PlanningInterfaceTestManager::testOffTrackFromPath(rclcpp::Node::SharedPtr
 void PlanningInterfaceTestManager::testOffTrackFromPathWithLaneId(
   rclcpp::Node::SharedPtr target_node)
 {
-  // write me
-  (void)target_node;
+  publishNominalPathWithLaneId(target_node, input_path_with_lane_id_name_);
+
+  const std::vector<double> deviation_from_path = {0.0, 1.0, 10.0, 100.0};
+  for (const auto & deviation : deviation_from_path) {
+    publishOdometry(target_node, input_odometry_name_, deviation);
+    test_utils::spinSomeNodes(test_node_, target_node, 5);
+  }
 }
 
 void PlanningInterfaceTestManager::testOffTrackFromTrajectory(rclcpp::Node::SharedPtr target_node)
 {
-  // write me
-  (void)target_node;
+  publishNominalTrajectory(target_node, input_trajectory_name_);
+
+  const std::vector<double> deviation_from_traj = {0.0, 1.0, 10.0, 100.0};
+  for (const auto & deviation : deviation_from_traj) {
+    publishOdometry(target_node, input_odometry_name_, deviation);
+    test_utils::spinSomeNodes(test_node_, target_node, 5);
```

Comment:
spinSomeNodes is executed in the publish odometry function [here](https://github.com/autowarefoundation/autoware.universe/blob/c0ebaf351103c8343021b3d1f7e013b74462cfae/planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp#LL368C3-L368C13)
so `spinSomeNodes` here is not necessary

Reply:
@kyoichi-sugahara In both `publishOdometry()` and `publishNominalTrajectory()`, spin is called only once. Considering the possibility that a topic may be dropped or may not be published due to topics stored in the queue before, it is better to call spin for sometimes. 
Actually, the topic drop often happens in the trajectory_follower_node testing.

Reply:
Addressed here: https://github.com/autowarefoundation/autoware.universe/pull/3587/commits/3fd8e01f65802f6d0e72b9770970885541cb25b5